packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +324,74 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("should not call entry point before data store is requested", async () => {
+		const dataStoreFactory1 = new DataObjectFactory(
+			TestDataObjectType1,
+			TestDataObject1,
+			[],
+			[],
+		);
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+		);
+		const loader = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+
+		// Create a container for the first client.
+		const container1 = await createAndAttachContainer(
+			provider.defaultCodeDetails,
+			loader,
+			provider.driver.createCreateNewRequest(provider.documentId),
+		);
+		assert.doesNotThrow(
+			async () => requestFluidObject<TestDataObject1>(container1, "/"),
+			"Initialization from container client should success.",
+		);
+
+		// Create a summarizer for the container and do a summary shouldn't throw.
+		const createSummarizerResult = await createSummarizerFromFactory(
+			provider,
+			container1,
+			dataStoreFactory1,
+			undefined,
+			containerRuntimeFactoryWithDefaultDataStore,
+			registryStoreEntries,
+		);
+		const summarizer = createSummarizerResult.summarizer;
+		assert.doesNotThrow(() => {
+			summarizer.summarizeOnDemand({ reason: "test" });
+		});
+
+		// In summarizer, load the data store should fail.
+		const summarizerContainer = await provider.loadContainer(runtimeFactory);
+		await assert.rejects(
+			async () => requestFluidObject<TestDataObject1>(summarizerContainer, "/"),
+
+			"This validates the summarizer client doesn't fully initialize the data object.",
+		);
+
+		// Load second container, load the data store again should success.
+		const url = (await container1.getAbsoluteUrl("/")) as string;
+		const container2 = await loader.resolve({ url });
+		assert.doesNotThrow(
+			async () => requestFluidObject<TestDataObject1>(container2, "/"),
+			(e) => e.message === "EntryPoint not implemented",
+			"This validates that an interactive client doesn't fully initialize the data object.",
+		);
```

Comment:
I'm a bit confused here... I would expect the `requestFluidObject()` call to throw, because when loading an existing container (vs creating a new one), the data store's `initializingFromExisting()` method should be called (vs `initializingFirstTime()`), so loading the container and requesting the data store should fail. I suspect this might actually be returning the same container created above (the loader probably has it cached so `loader.resolve()` just returns it) instead of loading a new one.

Let's keep this call but I think we should just extract the `requestFluidObject()` call out of the `assert.doesNotThrow`; if it throws, let the uncaught error be what makes the test fail, and let's update the comment above this section with something like `// Letting the loader resolve the same container from cache should still not throw when requesting the data store because it returns the container that already initialized fully when it was created`. And let's add another section below similar to this one, where we tell the loader to *not* give us a cached container, to load a new one instead (`await loader.resolve({ url, headers: { LoaderHeader.cache : false } });` needs some syntax fix or building the request in a few steps), and where we do expect `requestFluidObject()` to throw.

Reply:
Also, like Alex said, it's not a good idea to re-use the loader. It's more realistic to use a new loader and provider.loadContainer() will do that for you.

Reply:
To make this validation pass, the initializingFromExisting should look something like this:
```
protected async initializingFromExisting(): Promise<void> {
  if (this.context.clientDetails.capabilities.interactive === false) {
    throw new Error("EntryPoint not implemented");
  }
}
```

Reply:
Make sense. Thanks for guiding me.